frontends/web/src/components/transactions/transaction.tsx

```diff
@@ -14,22 +14,53 @@
  * limitations under the License.
  */
 
-import { Component, h } from 'preact';
-import { translate } from 'react-i18next';
-import { FiatConversion } from '../rates/rates';
+import { Component, h, RenderableProps } from 'preact';
+import { translate, TranslateProps } from '../../decorators/translate';
+import { Fee } from '../../routes/account/send/send';
 import A from '../anchor/anchor';
-import * as style from './transaction.css';
-import * as parentStyle from './transactions.css';
 import { Dialog } from '../dialog/dialog';
 import { ProgressRing } from '../progressRing/progressRing';
+import { FiatConversion } from '../rates/rates';
+import { Amount } from '../rates/rates';
+import { ArrowIn, ArrowOut, ArrowSelf, ExpandIcon } from './components/icons';
+import * as style from './transaction.css';
+import * as parentStyle from './transactions.css';
+
+interface State {
+    transactionDialog: boolean;
+}
+
+interface TransactionProps {
+    index: number;
+    explorerURL: string;
+    type: 'send' | 'receive' | 'self';
+    txID: string;
+    amount: Amount;
+    fee: Fee;
+    feeRatePerKb: Fee;
+    gas: number;
+    vsize: number;
+    size: number;
+    weight: number;
+    numConfirmations: number;
+    numConfirmationsComplete: number;
+    time: string;
```

Comment:
this can be also null or undefined (not sure which one) on new tx that have not been verified yet (that's why it's accessed with `time ? ... : ...` below.